File: app/src/main/java/com/example/newbiechen/ireader/ui/adapter/CategoryAdapter.java
package com.example.newbiechen.ireader.ui.adapter;

import android.view.View;
import android.view.ViewGroup;

import com.example.newbiechen.ireader.ui.adapter.view.CategoryHolder;
import com.example.newbiechen.ireader.ui.base.EasyAdapter;
import com.example.newbiechen.ireader.ui.base.adapter.IViewHolder;
import com.example.newbiechen.ireader.widget.page.TxtChapter;

/**
 * Created by newbiechen on 17-6-5.
 */

public class CategoryAdapter extends EasyAdapter<TxtChapter> {
    private int currentSelected = 0;
    @Override
    protected IViewHolder<TxtChapter> onCreateViewHolder(int viewType) {
        return new CategoryHolder();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = super.getView(position, convertView, parent);
        CategoryHolder holder = (CategoryHolder) view.getTag();

        if (position == currentSelected){
            holder.setSelectedChapter();
        }

        return view;
    }

    public void setChapter(int pos){
        currentSelected = pos;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/newbiechen/ireader/ui/adapter/CategoryAdapter.kt
package com.example.newbiechen.ireader.ui.adapter

import android.widget.TextView
import androidx.core.content.ContextCompat
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.example.newbiechen.ireader.App
import com.example.newbiechen.ireader.R
import com.example.newbiechen.ireader.utils.BookManager
import com.example.newbiechen.ireader.widget.page.TxtChapter

class CategoryAdapter(data: List<TxtChapter>) : BaseQuickAdapter<TxtChapter, BaseViewHolder>(R.layout.item_category, data) {
    override fun convert(helper: BaseViewHolder, item: TxtChapter) {
        //TODO:目录显示设计的有点不好，需要靠成员变量是否为null来判断。
        //首先判断是否该章已下载
        //如果没有链接地址表示是本地文件
        val drawable = if (item.link.isEmpty()) {
            ContextCompat.getDrawable(App.getInstance(), R.drawable.selector_category_load)
        } else {
            if (item.bookId.isNotBlank() && BookManager.isChapterCached(item.bookId, item.title)) {
                ContextCompat.getDrawable(App.getInstance(), R.drawable.selector_category_load)
            } else {
                ContextCompat.getDrawable(App.getInstance(), R.drawable.selector_category_unload)
            }
        }

        val category_tv_chapter = helper.getView<TextView>(R.id.category_tv_chapter)
        if (item.isSelected) {
            category_tv_chapter.setTextColor(ContextCompat.getColor(App.getInstance(), R.color.light_red))
        } else {
            category_tv_chapter.setTextColor(ContextCompat.getColor(App.getInstance(), R.color.nb_text_default))
        }
        category_tv_chapter.setCompoundDrawablesWithIntrinsicBounds(drawable, null, null, null)
        category_tv_chapter.text = item.title
    }

}